pkg/controller/import-controller.go

```diff
@@ -803,6 +806,58 @@ func (r *ImportReconciler) getVddkImageName() (*string, error) {
 	return nil, errors.Errorf("found %s ConfigMap in namespace %s, but it does not contain a '%s' entry", common.VddkConfigMap, namespace, common.VddkConfigDataKey)
 }
 
+func (r *ImportReconciler) updatePVCBoundContion(pvc *corev1.PersistentVolumeClaim) {
+	// set bound condition by getting the latest event
+	events := &corev1.EventList{}
+
+	err := r.client.List(context.TODO(), events,
+		client.InNamespace(pvc.GetNamespace()),
+		client.MatchingFields{"involvedObject.name": pvc.GetName(),
+			"involvedObject.uid": string(pvc.GetUID())},
+	)
+
+	if err != nil || len(events.Items) == 0 {
+		return
+	}
+
+	// Sort event lists by most recent
+	sort.Slice(events.Items, func(i, j int) bool {
+		return events.Items[i].FirstTimestamp.Time.After(events.Items[j].FirstTimestamp.Time)
```

Comment:
Would it make sense to sort on both event.Message containing and the time? Then if you sort it so the newest event with pvcPrime in the message is the top element, you don't have to loop anymore. You can just get the first item.